packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -138,6 +146,9 @@ export class CoreVersionedRoute implements VersionedRoute {
       validation?.request &&
       Boolean(validation.request.body || validation.request.params || validation.request.query)
     ) {
+      if (this.enableQueryVersion) {
```

Comment:
Thinking out loud, but I wonder if it wouldn't make sense to have a more explicit error when the query version is provided but `enableQueryVersion` is `false`: ATM we will just pass the version down to the handler, which will very likely throw a schema validation error from the params (or silently ignore it if there's not query param validation registered on the handler).

Are we fine with this, or would it be better to throw earlier here with a more explicit error message (e.g `passing the version via query param is not allowed on this API, please use the header instead...` or something)

What do you think about it?

Reply:
> enableQueryVersion is false: ATM we will just pass the version down to the handler

Yeah I was thinking of this case as "business as usual". If enable query version is false (the default) we will not reserve any keys on the query.

Stated differently: setting query version to true is the only way your handlers will never get apiVersion passed to them.

From what you're describing we should reserve that key across all endpoints?

Reply:
Yeah, we could think about just declaring this specific query param as a reserved one, unconditionally (I somehow doubt it would be an issue). 

But the impact either way is minimal, I'm not sure to have a preference. As said, it was mostly me writing my thoughts down :)

Reply:
I think that could make sense. It would prevent a situation where, one day, someone wants to turn on the feature but it using that key already 🤷🏼‍♂️. Let's see what happens if we make it an error condition